lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -655,12 +669,24 @@ int32_t nrf_modem_os_trace_put(const uint8_t * const data, uint32_t len)
 	uint32_t remaining_bytes = len;
 
 	while (remaining_bytes) {
-		uint8_t transfer_len = MIN(remaining_bytes, RTT_BUF_SZ);
 		uint32_t idx = len - remaining_bytes;
-
-		SEGGER_RTT_WriteSkipNoLock(trace_rtt_channel, &data[idx],
-			transfer_len);
+		uint32_t transfer_len;
+
+		transfer_len = SEGGER_RTT_WriteNoLock(trace_rtt_channel, &data[idx],
+							remaining_bytes);
+		if (transfer_len == 0) {
+			/* The API failed to write any bytes into the RTT buffer. This could be
+			 * because the RTT listener is not running on host side or the RTT speed
+			 * configured on host side is too slow. In this situation, we stop
+			 * attempting to write more bytes of the current trace and break out of
+			 * the loop. The trace bytes that have not been written are considered
+			 * lost.
+			 */
+			break;
+		}
 		remaining_bytes -= transfer_len;
+
+		rtt_read_delay();
```

Comment:
yeah, especially seeing that the delay is called on each write, I think we should rather let the kernel reschedule instead of busy-looping.